examples/ties.c

```diff
@@ -0,0 +1,359 @@
+//
+// Created by Mikael Vejdemo Johansson on 4/7/15.
+//
+// Intention: read in a parser, generate the system of equations for its
+// generating functions
+//
+
+#include <inttypes.h>
+#include "../src/backends/contextfree.h"
+#include "../src/backends/lr.h"
+#include <stdio.h>
+
+static const char *nonterminal_name(const HCFGrammar *g, const HCFChoice *nt)
```

Comment:
I'd like to avoid duplicating code. Since this function and `void readsequence(...)` both appear in both examples, can you please refactor them up into a separate header/implementation and `#include` it into the examples?